src/makeCalculator.js

```diff
@@ -37,7 +37,39 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  const calculator = {};
+
+  calculator.result = 0;
+
+  calculator.add = function(a, b) {
+    return a + b;
+  };
+
+  calculator.subtract = function(a, b) {
+    return a - b;
+  };
+
+  calculator.multiply = function(a, b) {
+    return a * b;
+  };
+
+  calculator.divide = function(a, b) {
+    return a / b;
+  };
+
+  calculator.reset = function() {
+    calculator.result = 0;
+
+    return this;
+  };
+
```

Comment:
It is a possible solution, but why you don't write all these methods inside your object? It will allow you to use `this`.

Reply:
I made like this because Ihor asked to make that way, i forgot the correct name of this, but the point is that we take properties only inside function. @ivan-kadykalo